pkg/services/contexthandler/contexthandler.go

```diff
@@ -206,29 +251,19 @@ func (h *ContextHandler) initContextWithAPIKey(reqContext *models.ReqContext) bo
 	defer span.End()
 
 	// base64 decode key
-	decoded, err := apikeygen.Decode(keyString)
-	if err != nil {
-		reqContext.JsonApiErr(401, InvalidAPIKey, err)
-		return true
-	}
-
-	// fetch key
-	keyQuery := models.GetApiKeyByNameQuery{KeyName: decoded.Name, OrgId: decoded.OrgId}
-	if err := h.SQLStore.GetApiKeyByName(reqContext.Req.Context(), &keyQuery); err != nil {
-		reqContext.JsonApiErr(401, InvalidAPIKey, err)
-		return true
+	// We always need to try old key format first otherwise we'll invalidate all keys
+	// beginning with "gl_"
```

Comment:
Realistically, would it be possible for legacy keys to ever start with `gl_`? Would it make sense to check for `{prefix}{serviceId}_` instead (to further disambiguate) and sidestep the overhead of trying to decode keys when we know they will not be valid?

Reply:
you're right, due to the way the old tokens are encoded it's unlikely they'll begin with something different than the encoding of `{'`. Great catch